cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -248,6 +256,42 @@ def _get_data(self):
             )
             return False
 
+    def _crawl_metadata(self):
+        start_time = time.monotonic()
+
+        self.userdata_raw = ec2.get_instance_userdata(
+            self.api_ver, self.metadata_address
+        )
+        LOG.debug(self.userdata_raw)
+        self.metadata = ec2.get_instance_metadata(
+            self.api_ver, self.metadata_address
+        )
+        LOG.debug(self.metadata)
+        LOG.debug(
+            "Crawl of metadata service took %s seconds",
+            int(time.monotonic() - start_time),
+        )
+
+        password_client = CloudStackPasswordServerClient(self.vr_addr)
+        try:
+            set_password = password_client.get_password()
```

Comment:
Unrelated, I'm interested in also timing the performance of get_password here as the network round trip contributes to the overall cost of _get_data(and the crawl_metadata method),

```diff
@@ -66,6 +66,7 @@ class CloudStackPasswordServerClient:
         )
         return output.strip()
 
+    @performance.timed("Getting password", log_mode="always")
     def get_password(self):
         password = self._do_request("send_my_password")
         if password in ["", "saved_password"]:
```